src/background/utils/promise-barrier.ts

```diff
@@ -0,0 +1,32 @@
+export class PromiseBarrier {
+    private resolves = [];
+    private rejects = [];
```

Comment:
```suggestion
    private rejects: Array<Promise<void>> = [];
```

Reply:
Done!

Since this is an array of `reject` arguments from `Promise` constructor (as opposed to array of Promises), it I made it `Array<(value?: any) => void>`.